pymeasure/instruments/hp/hp437b.py

```diff
@@ -0,0 +1,436 @@
+    74: "Display not responding",
+    75: "Digital failure",
+}
+
+
+class StatusMessage:
+    MeasurementErrorCode = (0, 1)
+    EntryErrorCode = (2, 2)
+    OperatingMode = (4, 2)
+    Range = (6, 2)
+    # 8, 9 unused
+    AutoFilterStatus = (10, 1)
+    Filter = (11, 1)
+    # 12, 13 unused
+    LinearLogStatus = (14, 1)
+    # A
+    PowerRefStatus = (16, 1)
+    RelativeModeStatus = (17, 1)
+    TriggerMode = (18, 1)
+    GroupTriggerMode = (19, 1)
+    LimitsCheckingStatus = (20, 1)
+    LimitsStatus = (21, 1)
+    # 22 unused
+    OffsetStatus = (23, 1)
+    DutyCycleStatus = (24, 1)
+    MeasurementUnits = (25, 1)
+
+
+class MeasurementUnit(Enum):
+    Watts = 0
+
+    dBm = 1
+
+    Percent = 2
+
+    dB = 3
+
+
+class HP437B(Instrument):
+    """Represents the HP856XX series spectrum analyzers.
+
+    Don't use this class directly - use their derivative classes
+
+    .. note::
+        Most command descriptions are taken from the document:
+        'HP 8560A, 8561B Operating & Programming'
+    """
+
+    def __init__(self, adapter, name="Hewlett-Packard HP437B", **kwargs):
+        super().__init__(
+            adapter,
+            name,
+            includeSCPI=True,
+            send_end=True,
+            **kwargs,
+        )
+
+    def __getstatus(status_type: StatusMessage, modifier=lambda v: v):
+        start_index, stop_offset = status_type
+        return lambda v: modifier(int(v[start_index:start_index + stop_offset]))
+
+    @property
+    def options(self):
+        raise NotImplementedError("This instrument is only partial SCPI conformant")
+
+    def check_errors(self):
+        errors = []
+        while True:
+            err = self.values("ERR?")
+            # exclude upper limit and lower limit hit from real errors
+            if int(err[0]) != 0 and int(err[0]) != 21 and int(err[0]) != 23:
+                log.error(f"{self.name}: {err[0]}, {Errors[err[0]]}")
+                errors.append(err)
+            else:
+                break
+        return errors
+
+    event_status = Instrument.measurement(
+        "*ESR?",
+        """
+        Get the status byte and Master Summary Status bit.
+        
+        .. code-block:: python
+            
+            print(instr.request_service_conditions)
+            StatusRegister.PowerOn|CommandError
+        """,
+        get_process=lambda v: EventStatusRegister(int(v))
+    )
+
+    def activate_auto_range(self):
+        """
+        The power meter divides each sensor’s power range into 5 ranges of
+        10 dB each. Range 1 is the most sensitive (lowest power levels), and
+        Range 5 is the least sensitive (highest power levels). Range 5 can be
+        less than 10 dB if the sensor’s power range is less than 50 dB. The
+        range can be set either automatically or manually.
+        'activate_auto_range' automatically selects the correct range for the current
+        measurement.
+        """
+        self.write("RA")
```

Comment:
Please keep the docstrings in the format proposed in the [style guide](https://pymeasure.readthedocs.io/en/latest/dev/coding_standards.html#docstrings)